### DISCUSSION TITLE
Outputing the residual as an AuxVariable?

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#26897] [Q&A General] [answered]
Hi all.

So we are having problem with convergence from our hydro simulation,

just wondering, is it possible to output the residual as an auxVariable? 

That way we could find "where" the problem is (check via Paraview), whether it appears to be a spatially local area, or overall bad residuals

![image](https://github.com/idaholab/moose/assets/79562439/6b310755-4764-4a41-a917-501375c24ddd)

Kind regards,
Traiwit

**Comment by Kavan-Khaledi** [comment, marked as answer]
Hello, you can use DebugResidualAux for this:
https://mooseframework.inl.gov/source/auxkernels/DebugResidualAux.html

Best regards,
Kavan

**Reply by Traiwit** [reply]
this is what we need, thank you @Kavan-Khaledi !
